### PR TITLE
Add section about draft PRs to handbook

### PR DESCRIPTION
**Pull request by christinaforney**
First commit is auto-formatting from prettier, second commit is added section.

### CODE REVIEW

File: handbook/engineering/code_reviews.md [hunk 1]
```diff
@@ -108,3 +108,11 @@ If your change touches multiple parts of our codebase (e.g. Go, TypeScript), the
 GitHub will automatically assign reviewers if there is a matching entry in the [CODEOWNERS](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/.github/CODEOWNERS) file, but that doesn't necessarily mean that you need to wait for an approval from everyone. For example, if you are making a change to the search backend then you only need approval from one person on that team, not all of them.
 
 Use your best judgement and ask if you are uncertain.
+
+## When should I use a draft PR?
+
+A draft PR signals that the change is not ready for reviewed. If you are looking for feedback or discussion on your change, you should mark the PR as ready for review, even if you expect more work to be done on the change.
```
Comment (nicksnyder) [inline handbook/engineering/code_reviews.md]: Suggest making this change and then deleting the bullet below.

```suggestion
A draft PR signals that the change is not ready for reviewed. This is useful, for example, if you want to self-review your diff before sending review requests to others. If you are looking for feedback or discussion on your change, then you should mark the PR as ready for review and communicate your intentions in the PR description.
```

File: handbook/engineering/code_reviews.md [hunk 2]
```diff
@@ -108,3 +108,11 @@ If your change touches multiple parts of our codebase (e.g. Go, TypeScript), the
...
+
+Draft PRs are great for:
```
Comment (christinaforney) [inline handbook/engineering/code_reviews.md]:
```suggestion
```

File: handbook/engineering/code_reviews.md [hunk 3]
```diff
@@ -108,3 +108,11 @@ If your change touches multiple parts of our codebase (e.g. Go, TypeScript), the
...
+
```
Comment (christinaforney) [inline handbook/engineering/code_reviews.md]:
```suggestion
```

File: handbook/engineering/code_reviews.md [hunk 4]
```diff
@@ -108,3 +108,11 @@ If your change touches multiple parts of our codebase (e.g. Go, TypeScript), the
...
+- Doing a self-review (see [effective code reviews](#what-makes-an-effective-code-review) above).
```
Comment (christinaforney) [inline handbook/engineering/code_reviews.md]:
```suggestion
```

**Pull request merged**